core/eolearn/core/utilities/other.py

```diff
@@ -0,0 +1,199 @@
+
+import uuid
+import numpy as np
+import geopandas as gpd
+from geopandas.testing import assert_geodataframe_equal
+
+from ..exceptions import EODeprecationWarning
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class LogFileFilter(Filter):
+    """Filters log messages passed to log file."""
+
+    def __init__(self, thread_name: Optional[str], *args, **kwargs):
+        """
+        :param thread_name: Name of the thread by which to filter logs. By default it won't filter by any name.
+        """
+        self.thread_name = thread_name
+        super().__init__(*args, **kwargs)
+
+    def filter(self, record: LogRecord):
+        """Shows everything from the thread that it was initialized in."""
+        return record.threadName == self.thread_name
+
+
+def deep_eq(fst_obj, snd_obj):
+    """Compares whether fst_obj and snd_obj are deeply equal.
+
+    In case when both fst_obj and snd_obj are of type np.ndarray or either np.memmap, they are compared using
+    np.array_equal(fst_obj, snd_obj). Otherwise, when they are lists or tuples, they are compared for length and then
+    deep_eq is applied component-wise. When they are dict, they are compared for key set equality, and then deep_eq is
+    applied value-wise. For all other data types that are not list, tuple, dict, or np.ndarray, the method falls back
+    to the __eq__ method.
+
+    Because np.ndarray is not a hashable object, it is impossible to form a set of numpy arrays, hence deep_eq works
+    correctly.
+
+    :param fst_obj: First object compared
+    :param snd_obj: Second object compared
+    :return: `True` if objects are deeply equal, `False` otherwise
+    """
+    # pylint: disable=too-many-return-statements
+    if not isinstance(fst_obj, type(snd_obj)):
+        return False
+
+    if isinstance(fst_obj, np.ndarray):
+        if fst_obj.dtype != snd_obj.dtype:
+            return False
+        fst_nan_mask = np.isnan(fst_obj)
+        snd_nan_mask = np.isnan(snd_obj)
+        return np.array_equal(fst_obj[~fst_nan_mask], snd_obj[~snd_nan_mask]) and np.array_equal(
+            fst_nan_mask, snd_nan_mask
+        )
+
+    if isinstance(fst_obj, gpd.GeoDataFrame):
+        try:
+            assert_geodataframe_equal(fst_obj, snd_obj)
+            return True
+        except AssertionError:
+            return False
+
+    if isinstance(fst_obj, (list, tuple)):
+        if len(fst_obj) != len(snd_obj):
+            return False
+
+        for element_fst, element_snd in zip(fst_obj, snd_obj):
+            if not deep_eq(element_fst, element_snd):
+                return False
+        return True
+
+    if isinstance(fst_obj, dict):
+        if fst_obj.keys() != snd_obj.keys():
+            return False
+
+        for key in fst_obj:
+            if not deep_eq(fst_obj[key], snd_obj[key]):
+                return False
+        return True
+
+    return fst_obj == snd_obj
+
+
+def constant_pad(X, multiple_of, up_down_rule="even", left_right_rule="even", pad_value=0):
```

Comment:
Maybe we can rename this `X` to `array` or `image` or something similar? This way we get rid of `# pylint: disable=invalid-name`.